lib/uplinkc/uplink_definitions.h

```diff
@@ -70,7 +91,48 @@ typedef struct EncryptionAccess {
     char key[32];
 } EncryptionAccess;
 
+typedef struct ObjectInfo {
+    uint32_t   version;
+    BucketInfo bucket;
+    char       *path;
+    bool       is_prefix;
+    char       *content_type;
+    int64_t    created;
+    int64_t    modified;
+    int64_t    expires;
+} ObjectInfo;
+
+typedef struct ObjectList {
+    char       *bucket;
+    char       *prefix;
+    bool       more;
+    ObjectInfo *items;
+    int32_t    length;
+} ObjectList;
+
 typedef struct UploadOptions {
     char    *content_type;
     int64_t expires;
 } UploadOptions;
+
+typedef struct ListOptions {
+    char    *prefix;
+    char    *cursor;
+    char    delimiter;
+    bool    recursive;
+    int8_t  direction;
+    int64_t limit;
+} ListOptions;
+
+typedef struct ObjectMeta {
+    char     *bucket;
+    char     *path;
+    bool     is_prefix;
```

Comment:
Is prefix doesn't make sense to me on a type that is not part of list.

Reply:
leave as-is